### ISSUE TITLE
Nested interactive controls are not announced by screen readers

### ISSUE THREAD

**Issue by msftedad** [ag-grid/ag-grid#6350]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```
**Providing a Reproducible Scenario
Repro Steps**
1.Open URL: https://o0tgbf.csb.app/
2.Navigate to fliter option by using Tab key and Press Enter.
3.Open accessibility insights for web tool and run fastpass.
4.Verify whether nested interactive controls are announced by screen readers or not.

**Current behavior**
Nested interactive controls are not announced by screen readers

**Expected behavior**
Nested interactive controls should be announced by screen readers

**Please tell us about your environment:**
OS Version: 22H2 OS Build 22621.1265
AG Grid version: 28.2.1
Browser: Edge(Chromium) Version 110.0.1587.63 (Official build) (64-bit)
Language: React js
![image](https://user-images.githubusercontent.com/93735775/226603105-549a5fba-a600-47ef-8977-e41739754e4f.png)

**Comment by kiril-matev** [comment]
Hello @msftedad,

Thanks for reporting this.

This is expected behavior because AG Grid is using a virtualised list to present the list of items in the set filter. This requires that a container element is placed between the filter element and the actual options themselves. This allows the filter to display thousands of set filter items while only using a few actual DOM elements which are reused to display different items when the set filter is scrolled. AG Grid implement custom keyboard navigation, focus handling logic and has the necessary accessibility attributes set to all the necessary elements to ensure they are announced correctly.

However, Accessibility audit tools such as Accessibility Insights for Web use static DOM analysis and assume that all elements of AG Grid are rendered in the DOM, which isn't the case due to virtualization. Even though AG Grid handles keyboard navigation, focus and accessibility announcements internally and these are correct in terms of the end-user experience, the accessibility audit tools report these as issues, while in fact when the user is using the grid these aren't actual issues.

Please use this justification to close the issue on your side as not having any impact on the end-user experience.

**Comment by msftedad** [comment]
@kiril-matev Thank you for your response. We will inform this to our engineering team and get back to you.
